spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -1189,91 +1191,48 @@ public InterpreterResult interpret(String[] lines, InterpreterContext context) {
-        boolean continuation = false;
-        if (nextLine.isEmpty()
-           || nextLine.startsWith("//")         // skip empty line or comment
-           || nextLine.startsWith("}")
-           || nextLine.startsWith("object")) {  // include "} object" for Scala companion object
-          continuation = true;
-        } else if (!inComment && nextLine.startsWith("/*")) {
-          inComment = true;
-          continuation = true;
-        } else if (inComment && nextLine.lastIndexOf("*/") >= 0) {
-          inComment = false;
-          continuation = true;
-        } else if (nextLine.length() > 1
-                && nextLine.charAt(0) == '.'
-                && nextLine.charAt(1) != '.'     // ".."
-                && nextLine.charAt(1) != '/') {  // "./"
-          continuation = true;
-        } else if (inComment) {
-          continuation = true;
-        }
-        if (continuation) {
-          incomplete += s + "\n";
-          continue;
-        }
-      }
 
-      scala.tools.nsc.interpreter.Results.Result res = null;
-      try {
-        res = interpret(incomplete + s);
-      } catch (Exception e) {
-        sc.clearJobGroup();
-        out.setInterpreterOutput(null);
-        logger.info("Interpreter exception", e);
-        return new InterpreterResult(Code.ERROR, InterpreterUtils.getMostRelevantMessage(e));
-      }
+    scala.tools.nsc.interpreter.Results.Result res;
+    try {
+      res = interpret(code);
 
-      r = getResultCode(res);
+      Code r = getResultCode(res);
 
       if (r == Code.ERROR) {
-        sc.clearJobGroup();
-        out.setInterpreterOutput(null);
         return new InterpreterResult(r, "");
-      } else if (r == Code.INCOMPLETE) {
-        incomplete += s + "\n";
-      } else {
-        incomplete = "";
       }
-    }
 
-    // make sure code does not finish with comment
-    if (r == Code.INCOMPLETE) {
-      scala.tools.nsc.interpreter.Results.Result res = null;
-      res = interpret(incomplete + "\nprint(\"\")");
-      r = getResultCode(res);
-    }
+      // make sure code does not finish with comment
+      if (r == Code.INCOMPLETE) {
+        res = interpret(code + "\nprint(\"\")");
+        r = getResultCode(res);
+      }
 
-    if (r == Code.INCOMPLETE) {
-      sc.clearJobGroup();
-      out.setInterpreterOutput(null);
-      return new InterpreterResult(r, "Incomplete expression");
-    } else {
-      sc.clearJobGroup();
-      putLatestVarInResourcePool(context);
+      if (r == Code.INCOMPLETE) {
+        return new InterpreterResult(r, "Incomplete expression");
+      } else {
+        putLatestVarInResourcePool(context);
+        return new InterpreterResult(Code.SUCCESS);
+      }
+    } catch (Exception e) {
+      logger.info("Interpreter exception", e);
+      return new InterpreterResult(Code.ERROR, InterpreterUtils.getMostRelevantMessage(e));
+    } finally {
+      try {
+        context.out.setType(InterpreterResult.Type.TEXT);
```

Comment:
I checked it:
![image](https://cloud.githubusercontent.com/assets/2759002/26236889/4cfaa28e-3c7b-11e7-89ec-fb2f3aed8c60.png)

"hello" is printed to `context.out` and REPL output is printed to temporary buffer. Then it is printed to `context.out` as a TEXT